pkg/sdk/warehouses.go

```diff
@@ -305,10 +305,32 @@ func (c *warehouses) Alter(ctx context.Context, id AccountObjectIdentifier, opts
 	if err != nil {
 		return err
 	}
+
+	// Warehouse needs to be suspended to change its type.
+	if opts.warehouseTypeIsChanged() {
+		warehouse, err := c.ShowByID(ctx, id)
+		if err != nil {
+			return err
+		}
+		if warehouse.State == WarehouseStateStarted {
+			err := c.Alter(ctx, id, &AlterWarehouseOptions{Suspend: Bool(true)})
+			if err != nil {
+				return err
+			}
+			defer func() {
+				_ = c.Alter(ctx, id, &AlterWarehouseOptions{Resume: Bool(true)})
```

Comment:
Maybe we should at least log an error when it occurred after a failed alter

Reply:
Oky

Reply:
Done.